### DISCUSSION TITLE
Shell segment to distinguish powershell from pwsh (core)

### DISCUSSION THREAD

**Discussion by PilotBob** [JanDeDobbeleer/oh-my-posh#6016] [Ideas]
I would like my shell segment to display pwsh when using PowerShell core and powershell when using PowerShell legacy 5.x. 

I can't seem to figure it out. AI suggested adding a condition: to the shell with a segment for each shell. But I couldn't find any docs on that. 

Is this something that could be modified in the segment itself to support?

**Comment by lewis-yeung** [comment]
@PilotBob You should use `oh-my-posh init powershell` in the profile of (Windows) PowerShell 5, and `oh-my-posh init pwsh` in that of PowerShell Core (e.g., version 7), the `shell` segment naturally shows different shell names for them.

**Reply by PilotBob** [reply]
ah brilliant. That works great for me. I copied the init from one to the other. :)

Thanks.
